particle-iot/device-os
# Fix/unit tests pre-releases

### Problem

Unit Tests do not expect Alpha or Beta pre-release strings in addition to RC.

### Solution

Give the Unit-Test some help in determining which type of string format is is testing

### Steps to Test

Run Unit Tests from the root directory (firmware $):
```
source ci/install_boost.sh
ci/build_boost.sh
cd user/tests/unit && make -s clean all
```

All tests should pass with these version strings:
```
1.2.0-alpha.1
1.2.0-beta.1
1.2.0-rc.1

===============================================================================
All tests passed (144673 assertions in 152 test cases)
```

---

### Completeness
